olp-cpp-sdk-core/src/client/PendingUrlRequests.h

```diff
@@ -61,6 +65,19 @@ class PendingUrlRequest {
   bool ExecuteOrCancelled(const ExecuteFuncType& func,
                           const CancelFuncType& cancel_func = nullptr);
 
+  /// Calls the managed CancellationContext ExecuteOrCancelled so that it can
+  /// remember the func returned CancellationToken to be able to cancel any
+  /// async operation taking place in our name. And then waits for response.
+  /// Default wait time is 60 seconds.
+  bool ExecuteAndWait(
```

Comment:
I am not sure if this is the right place for this. I would rather put it in OlpClient as it is more bound to the sync CallApi() method.

Reply:
It was for draft only. Completely removed